docs/source/references/api_spec.md

```diff
@@ -1,57 +1,59 @@
 # API Specification for Client Libraries
 
 - [API Specification for Client Libraries](#api-specification-for-client-libraries)
-  - [Endpoints](#endpoints)
+  - [Endpoints for API](#endpoints-for-api)
     - [Submit Instructions](#submit-instructions)
-    - [Submit Query](#submit-query)
+    - [Request a Query](#request-a-query)
```

Comment:
```suggestion
    - [Query](#query)
```
Little known fact is that query is both a verb and a noun. The verb query is what you often do to a query.